### PR TITLE
Fix false updates on cron resource when using integers

### PR DESCRIPTION
**Pull request by kimbernator**
Signed-off-by: Jeremy Kimber <kimbernator@state.local>

## Description
Coerce all integer values for cron into strings to avoid false updates 

## Related Issue

https://github.com/chef/chef/issues/12311

## Types of changes

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Chore (non-breaking change that does not add functionality or fix an issue)

## Checklist:

- [x] I have read the **CONTRIBUTING** document.
- [x] I have run the pre-merge tests locally and they pass.
- [x] I have updated the documentation accordingly.
- [x] I have added tests to cover my changes.
- [x] All new and existing tests passed.
- [x] All commits have been signed-off for [the Developer Certificate of Origin](https://github.com/chef/chef/blob/master/CONTRIBUTING.md#developer-certification-of-origin-dco).

### CODE REVIEW

File: lib/chef/provider/cron.rb [hunk 1]
```diff
@@ -88,7 +88,11 @@ def load_current_resource
 
       def cron_different?
         CRON_ATTRIBUTES.any? do |cron_var|
-          new_resource.send(cron_var) != current_resource.send(cron_var)
+          if new_resource.send(cron_var).is_a?(Integer)
+            new_resource.send(cron_var).to_s != current_resource.send(cron_var)
+          else
+            new_resource.send(cron_var) != current_resource.send(cron_var)
+          end
```
Comment (tpowell-progress) [inline lib/chef/provider/cron.rb:95]:
```suggestion
          new_resource.send(cron_var).to_s != current_resource.send(cron_var).to_s
```

Comment (kimbernator) [reply lib/chef/provider/cron.rb:95]: Problem with this is the `environment` attribute, which can be a hash. `to_s` won't take sorting into account.

Comment (marcparadise) [reply lib/chef/provider/cron.rb:95]: What if instead we check types - and only coerce them to string if they are not the same?

Comment (kimbernator) [reply lib/chef/provider/cron.rb]: Seems reasonable to me. Just pushed up a revision that will do that instead.

**Review by jaymzh — APPROVED** [review]

**Comment by jaymzh** [comment]
This looks good, can you rebase though, please? the builds have been fixed recently.

**Comment by jaymzh** [comment]
Ping @kimbernator - can you rebase please?

**Comment by tpowell-progress** [comment]
@kimbernator Now that GA has been released, can you please rebase so that we can confirm that tests still pass with these changes?

**Comment by johnmccrae** [comment]
@kimbernator Can you rebase and push this back up again please?

**Review by tpowell-progress — APPROVED** [review]

**Pull request merged**